### PR TITLE
Feature/azure data explorer managed identity

### PR DESCRIPTION
**Pull request by ag-ramachandran**
**Description:**
Add support for ManagedIdentity. This enables passwordless config for the ADX sink

**Link to tracking Issue:**  
#21924 

**Testing:** Manually tested

**Documentation:** Updated README documentation

### CODE REVIEW

File: exporter/azuredataexplorerexporter/adx_exporter.go [hunk 1]
```diff
@@ -205,8 +206,17 @@ func getMappingRef(config *Config, telemetryDataType int) ingest.FileOption {
...
-	kcsb := kusto.NewConnectionStringBuilder(config.ClusterURI).WithAadAppKey(config.ApplicationID, string(config.ApplicationKey), config.TenantID)
-	kcsb.SetConnectorDetails("OpenTelemetry", version, "", "", false, "", kusto.StringPair{})
+	var kcsb *kusto.ConnectionStringBuilder
+	isManagedIdentity := len(strings.TrimSpace(config.ManagedIdentityId)) == 0
+	// If the user has managed identity done, use it. For System managed identity use the MI as system
+	if isManagedIdentity && strings.ToUpper(strings.TrimSpace(config.ManagedIdentityId)) == "SYSTEM" {
+		kcsb = kusto.NewConnectionStringBuilder(config.ClusterURI).WithSystemManagedIdentity()
+	} else if isManagedIdentity {
+		kcsb = kusto.NewConnectionStringBuilder(config.ClusterURI).WithUserManagedIdentity(config.ManagedIdentityId)
+	} else {
```
[the remark this answers is not in the document]
Comment (ag-ramachandran) [reply exporter/azuredataexplorerexporter/adx_exporter.go:216]: Here was the thought we had, This condition will be checked in config before it reaches here

**Comment by ben-childs-docusign** [comment]
Thank you!

**Review by yogilad — APPROVED** [review]

**Comment by ag-ramachandran** [comment]
Hello @atoulme, @codeboten , @mx-psi , would this be something you would be able to look at when you get time for a review and approve. Thank you

**Comment by ben-childs-docusign** [comment]
Is there any chance of this getting reviewed and merged this week?

File: exporter/azuredataexplorerexporter/adx_exporter.go [hunk 2]
```diff
@@ -205,8 +206,19 @@ func getMappingRef(config *Config, telemetryDataType int) ingest.FileOption {
...
+	// If the user has managed identity done, use it. For System managed identity use the MI as system
+	switch {
+	case !isManagedIdentity:
+		kcsb = kusto.NewConnectionStringBuilder(config.ClusterURI).WithAadAppKey(config.ApplicationID, string(config.ApplicationKey), config.TenantID)
+	case isManagedIdentity && isSystemManagedIdentity:
+		kcsb = kusto.NewConnectionStringBuilder(config.ClusterURI).WithSystemManagedIdentity()
+	case isManagedIdentity && !isSystemManagedIdentity:
+		kcsb = kusto.NewConnectionStringBuilder(config.ClusterURI).WithUserManagedIdentity(config.ManagedIdentityID)
+	}
+	kcsb.SetConnectorDetails("OpenTelemetry", version, "", "", false, "", kusto.StringPair{Key: "isManagedIdentity", Value: strconv.FormatBool(isManagedIdentity)})
```
Comment (atoulme) [inline exporter/azuredataexplorerexporter/adx_exporter.go:221]: This switch statement is weird to me, can you add tests to cover it?

File: .chloggen/add_managed_identity_azuredataexplorer.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: azuredataexplorerexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add support for managed identity.This enables users to not use Key based authentication
```
Comment (atoulme) [inline .chloggen/add_managed_identity_azuredataexplorer.yaml:8]:
```suggestion
note: Add support for managed identity. This enables users to not use Key based authentication
```

**Comment by atoulme** [comment]
Please resolve comments if you have addressed them. I have added 2 comments.

The codeowners @asaharan and @ag-ramachandran need to review.

File: exporter/azuredataexplorerexporter/adx_exporter.go (hunk 2 above)
Comment (ag-ramachandran) [reply exporter/azuredataexplorerexporter/adx_exporter.go:226]: Added some tests for this

**Comment by ag-ramachandran** [comment]
> Please resolve comments if you have addressed them. I have added 2 comments.
> 
> The codeowners @asaharan and @ag-ramachandran need to review.

Good day @atoulme , Resolved all comments that were addressed. Thank you

**Review by atoulme — APPROVED** [review]
LGTM. Tests are rerunning.

**Pull request merged**